students/charlie_england/session07/html_render.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+
+"""
+A class-based system for rendering html.
+"""
+
+# This is the framework for the base class
+class Element(object):
+    tag = "html"
+
+    def __init__(self, content=None,**attributes):
+        self.content_lst = [f"<{self.tag}>",]
```

Comment:
hmm, I haven't seen this approach before.

But I think it's better to wait and add the tag in the render method -- this way, it is very fagile if the content os manipulated later on: the element is created without content, and content added later, or content removed (not part of the spec, but still...), etc.